planetiler-core/src/main/java/com/onthegomap/planetiler/ForwardingProfile.java

```diff
@@ -81,13 +81,21 @@ protected ForwardingProfile(Handler... handlers) {
   }
 
   private boolean caresAboutLayer(String layer) {
-    return (onlyLayers.isEmpty() || onlyLayers.contains(layer)) && !excludeLayers.contains(layer);
+    return ((onlyLayers.isEmpty() || onlyLayers.contains(layer)) && !excludeLayers.contains(layer)) ||
+      this.handlers.stream()
+        .filter(h -> h instanceof HandlerForLayer)
+        .map(h -> (HandlerForLayer) h)
+        .anyMatch(existing -> dependsOnLayer(existing.name(), layer));
   }
 
   public boolean caresAboutLayer(Object obj) {
     return !(obj instanceof HandlerForLayer l) || caresAboutLayer(l.name());
   }
 
+  public boolean dependsOnLayer(String dependent, String dependency) {
```

Comment:
onlyLayers and exluceLayers returns lists, would it make sense for this one to return a `Map<String, List<String>>` of the dependencies?

Reply:
Adjusted. Should I alse rename? Say `dependsOnLayer()` -> `layerDependencies()`?